### ISSUE TITLE
[Bug]: Widget doesnt evaluate; Dependency Map Issue

### ISSUE THREAD

**Issue by Tooluloope** [appsmithorg/appsmith#9096]
### Is there an existing issue for this?

- [ ] I have searched the existing issues

### Current Behavior

[![LOOM DEMO](http://cdn.loom.com/sessions/thumbnails/2051e9bc9a2b464a894e1a7531ecf924-00001.gif)](https://www.loom.com/share/2051e9bc9a2b464a894e1a7531ecf924)

From the loom video, I have this function in those widgets, and they all evaluate to undefined on page load, but when I remove their dependency from `input1.text` and make `const index = "IN"`, the evaluate correctly.
```js
{{function() {
    const index = Input1.text

    const obj = {
        "IN":[{
            "label":"Ezetap (BETA !!)",
            "value":"ACCESS_EZETAP"
            }], 
            "ZA":[{
            "label": "Yoco",
            "value": "ACCESS_YOCO"
            }],
            "NZ":[{
            "label": "Smartpay",
            "value":"ACCESS_SMARTAY"
            }],
            "AU":[{
            "label": "Tyro",
            "value": "ACCESS_TYRO"
            },{
            "label": "Smartpay",
            "value":"ACCESS_SMARTAY"
            }],
            "JP":[{
            "label": "Coiney",
            "value":"ACCESS_COINEY"
            },{"label": "CpayPro",
            "value":"ACCESS_ONEPAY"
            },{"label": "Softbank",
            "value":"ACCESS_SOFTBANK"
            },{"label": "Paygate",
            "value":"ACCESS_PAYGATE"
            }],
            "KR":[{
            "label": "Kiccgate",
            "value": "ACCESS_KICC"
            },{"label": "Nice",
            "value":"ACCESS_NICEPAY"
            }],
            "CA":[{
            "label": "Worldpay",
            "value":"ACCESS_VANTIV"
            }],
            "US":[
            {"label": "Worldpay",
            "value":"ACCESS_VANTIV"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"
            },{ "label": "Sumup",
            "value": "ACCESS_SUMUP"
            },{
            "label":"Cardconnect",
            "value":"ACCESS_FIRSTDATA"
            }],
            "BR":[{
            "label": "Sumup",
            "value":"ACCESS_SUMUP"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],
            "FI":[{
            "label": "Sumup",
            "value":"ACCESS_SUMUP"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],
            "GB":[{
            "label": "Sumup",
            "value":"ACCESS_SUMUP"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],
            "IT":[{
            "label": "Sumup",
            "value":"ACCESS_SUMUP"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],"NO":[{
            "label": "Sumup",
            "value":"ACCESS_SUMUP"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],"ES":[{
            "label": "Sumup",
            "value":"ACCESS_SUMUP"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],"NL":[{
            "label": "Sumup",
            "value":"ACCESS_SUMUP"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],
            "SE":[{
            "label": "Sumup",
            "value":"ACCESS_SUMUP"
            },{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],
            "DK":[{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],
            "MX":[{ "label": "iZettle",
            "value":"ACCESS_IZETTLE"}],
            "BE":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "CL":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "IE":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "AT":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "PT":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "PL":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"CH":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"CZ":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "GR":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "HU":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "LT":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"LV":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"SI":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"BG":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"CY":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"DK":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"EE":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"LU":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"MT":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"SK":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"RE":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"YT":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"HR":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"GN":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"GP":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"MQ":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"BL":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"PM":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"MF":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"GI":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"GU":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"PR":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"SJ":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"AX":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"GF":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"JE":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"IM":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],"GG":[{
            "label":"Sumup",
            "value":"ACCESS_SUMUP"
            }],
            "empty": []
    };

    return obj[index]

}()}}

```

### Steps To Reproduce

Create An Input widget, then give a default value of `IN` and then also create a select widget and add the function above to its options.
Then reload the page

### Environment

Production

### Version

Cloud

**Comment by eco-monk** [comment]
Bumped into the same issue yesterday. Created [an app](https://app.appsmith.com/applications/6194983843aaaf0225a8097f/pages/6194991843aaaf0225a80986) that reproduces this.


Note: If we create the Table before creating the Input widget. It seems to work fine. (atleast in my machine)
Try switching between the two pages after load. The first page depicts the error describe here.

**Comment by Rishabh-Rathod** [comment]
This issue is actually resolved with #9418. 

Also, the above-mentioned data has multiple "DX" keys in it which makes it an invalid Object, and our debugger shows an error for the same.
